### ISSUE TITLE
Full maintenance gets killed and never runs as scheduled

### ISSUE THREAD

**Issue by Zackyist** [kopia/kopia#1903]
**OS**: Ubuntu 20.04.4 LTS (headless)
**CPU**: 4 cores @ 2449.998 MHz of an AMD EPYC 7763
**RAM**: 8343 MB with an average of 6484 MB free
**Repo size**: 53 GB
**Kopia version**: 0.10.7

According to `kopia maintenance info` the scheduled full maintenance (set to the default 24h interval) is never run so my repo keeps growing in size without any pruning ever done. It always shows the next one as upcoming as scheduled but the list of last full maintenance runs only shows the ones I ran manually a long time ago. Snapshot GCs seem to run fine.

In addition, most of the time when trying to run full maintenance manually with `kopia maintenance run --full`, the process gets killed moments (or sometimes minutes) after starting it even when I have 6 GB RAM free and the CPU usage (before running) is consistently below 5 % according to *htop*. I believe this might be the reason the scheduled runs are failing as well.

Example of the process getting killed:
```
~ sudo kopia maintenance run --full
Running full maintenance...
Looking for active contents...
Looking for unreferenced contents...
Rewriting contents from short packs...
Killed
```

**Comment by CompuRoot** [comment]
`Killed` coming from operation system that don't like something. I saw such behavior on VPS where hypervisor killed `kopia` if I run there benchmarks

**Comment by Zackyist** [comment]
Yeah, I'm also running this on a VPS. Just tried a fresh run and had a look at the syslog. It seems like Kopia is using way too much memory (the system had 6,5 GB free before trying to run maintenance) - 11 GB if I read this correctly?

```
May  5 21:31:08 venom kernel: [804824.742330] kopia invoked oom-killer: gfp_mask=0x100dca(GFP_HIGHUSER_MOVABLE|__GFP_ZERO), order=0, oom_score_adj=0
May  5 21:31:08 venom kernel: [804824.742335] CPU: 1 PID: 838410 Comm: kopia Not tainted 5.4.0-109-generic #123-Ubuntu
May  5 21:31:08 venom kernel: [804824.742611] [ 838408]     0 838408  2808629  1643339 13860864        0             0 kopia
May  5 21:31:08 venom kernel: [804824.742619] oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=/,mems_allowed=0,global_oom,task_memcg=/user.slice/user-1000.slice/session-2329.scope,task=kopia,pid=838408,uid=0
May  5 21:31:08 venom kernel: [804824.742643] Out of memory: Killed process 838408 (kopia) total-vm:11234516kB, anon-rss:6573356kB, file-rss:0kB, shmem-rss:0kB, UID:0 pgtables:13536kB oom_score_adj:0
May  5 21:31:08 venom kernel: [804824.926011] oom_reaper: reaped process 838408 (kopia), now anon-rss:0kB, file-rss:0kB, shmem-rss:0kB
```

**Comment by CompuRoot** [comment]
@Zackuist

What compression are you using? I read [in issue 1233](https://github.com/kopia/kopia/issues/1233) that it happened when zstd used. Also can you try to run kopia with [this suggestion](https://github.com/kopia/kopia/issues/1233#issuecomment-941952447) because OS reports about memory might be very confusing.

**Comment by Zackyist** [comment]
I changed the compression method to *gzip* which seems to be the least memory intensive one according to the documentation but the process still gets killed when I have 6 GB of RAM free to begin with. I tried to use the *pprof* method to get the heap data while running `kopia --metrics-listen-addr 0.0.0.0:6060 --enable-pprof maintenance run` but I only got the following:

```
$ curl -LO http://localhost:6060/debug/pprof/heap
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    19  100    19    0     0  19000      0 --:--:-- --:--:-- --:--:-- 19000
```

Since the heap file is only 19 bytes, it seems like something's going wrong and the *Go* commands of course didn't work on the invalid data:

```
$ go tool pprof -alloc_objects -lines -pdf heap 
heap: parsing profile: unrecognized profile format
failed to fetch any source profiles
```

**Comment by nh2** [comment]
Linking my recent similar experience: https://github.com/kopia/kopia/issues/1233#issuecomment-1221642613

**Comment by nh2** [comment]
> Since the heap file is only 19 bytes

Those 19 bytes are `404 page not found`.

PR to fix that: https://github.com/kopia/kopia/pull/2339
